src/coreComponents/mesh/generators/VTKUtilities.cpp

```diff
@@ -190,72 +196,143 @@ vtkSmartPointer< vtkMultiProcessController > getController()
-    //       This removes duplicate points, either present in the dataset, or resulting from merging pieces.
-  }
-  else if( extension == "pvts" )
+  auto const serialRead = [&]( auto const vtkGridReader )
   {
-    auto const vtkSgReader = vtkSmartPointer< vtkXMLPStructuredGridReader >::New();
-    vtkSgReader->SetFileName( filePath.c_str() );
-    vtkSgReader->UpdateInformation();
-    vtkSgReader->UpdatePiece( MpiWrapper::commRank(), MpiWrapper::commSize(), 0 );
-    loadedMesh = vtkSgReader->GetOutput();
-  }
-  else
+    vtkGridReader->SetFileName( filePath.c_str() );
+    vtkGridReader->Update();
+    return vtkGridReader->GetOutput();
+  };
+
+  switch( stringToVTKMeshExtension( extension ) )
   {
-    if( MpiWrapper::commRank() == 0 )
+    case VTKMeshExtension::vtk:
     {
-      auto const read = [&]( auto const vtkGridReader )
-      {
-        vtkGridReader->SetFileName( filePath.c_str() );
-        vtkGridReader->Update();
-        return vtkGridReader->GetOutput();
-      };
-
-      if( extension == "vtk" )
-      {
-        loadedMesh = read( vtkSmartPointer< vtkUnstructuredGridReader >::New() );
-      }
-      else if( extension == "vtu" )
-      {
-        loadedMesh = read( vtkSmartPointer< vtkXMLUnstructuredGridReader >::New() );
-      }
-      else if( extension == "vts" )
-      {
-        loadedMesh = read( vtkSmartPointer< vtkXMLStructuredGridReader >::New() );
-      }
-      else if( extension == "vti" )
-      {
-        vtkSmartPointer< vtkImageData > image = read( vtkSmartPointer< vtkXMLImageDataReader >::New() );
-        vtkNew< vtkImageDataToPointSet > imageDataToSg;
-        imageDataToSg->SetInputData( image );
-        imageDataToSg->Update();
-        vtkStructuredGrid * sGrid = imageDataToSg->GetOutput();
-        loadedMesh = sGrid;
-      }
-      else
-      {
-        GEOSX_ERROR( extension << " is not a recognized extension for VTKMesh. Please use .vtk, .vtu, .vts, .vti, .pvtu or .ptvs." );
-      }
+      loadedMesh = MpiWrapper::commRank() == 0
+                 ? serialRead( vtkSmartPointer< vtkDataSetReader >::New() )
+                 : vtkSmartPointer< vtkUnstructuredGrid >::New().GetPointer();
+      //if( MpiWrapper::commRank() == 0 )
+      //{
+      //  loadedMesh = serialRead( vtkSmartPointer< vtkDataSetReader >::New() );
+      //}
+      //else
+      //{
+      //  loadedMesh = vtkSmartPointer< vtkUnstructuredGrid >::New();
+      //}
+      break;
     }
-    else
+    case VTKMeshExtension::vtu:
+    {
+      loadedMesh = MpiWrapper::commRank() == 0
+                 ? serialRead( vtkSmartPointer< vtkXMLUnstructuredGridReader >::New() )
+                 : vtkSmartPointer< vtkUnstructuredGrid >::New().GetPointer();
+      break;
+    }
+    case VTKMeshExtension::vtr:
+    {
+      loadedMesh = MpiWrapper::commRank() == 0
+                 ? serialRead( vtkSmartPointer< vtkXMLRectilinearGridReader >::New() )
```

Comment:
BTW, in a same way, is there an equivalent to the generic `vtkDataSetReader`? Maybe `vtkXMLDataReader`?

Reply:
I was not able to find a way to do this. Actually I gave up on using the generic `vtkDataSetReader` even for legacy vtk formats and ended up calling the specific readers (`vtkStructuredPointsReader`, `vtkStructuredGridReader`, `vtkUnstructuredGridReader`, `vtkRectilinearGridReader`).